### ISSUE TITLE
Integrate Azure.Provisioning into Aspire (CDK)

### ISSUE THREAD

**Issue by mitchdenny** [dotnet/aspire#2423]
We want to explore the integration of the types defined in `Azure.Provisioning` into `Aspire.Hosting.Azure`. This would be an alternative way of producing Bicep that we pass through to **azd** and should be built in parallel to the existing `BicepResource` and supporting extension methods.

Goals are:

- [x] Enable adding a CDK-based construct/constructs to the Aspire application model as a resource.
- [x] Integrate CDK-based resources into the existing parameter passing mechanism.
- [x] Integrate CDK-based resources into deployment via **azd** and the Aspire.Hosting.Azure.Provisioning infrastructure
- [x] Support getting outputs and secret outputs out of the CDK resource deployment to support chaining.
- [ ] Consider scenarios where we generate the entirety of the `main.bicep` file for AZD using the full context we have in the app model.

For this initial piece of work we are more concerned about getting the base abstraction right so that we are confident that Azure resources that we already have could be replatformed on top but we don't want to try and do all these resources at once (this is a side-by-side/parallel approach).

**Comment by mitchdenny** [comment]
Initial PR exploring this was: https://github.com/dotnet/aspire/pull/2252

The ground moved under that PR quite a bit so rather than fight with a merge that doesn't get us closer to where we want to go I abandoned it. I will be standing up another PR.

**Comment by mitchdenny** [comment]
Marking this as done. I think the work to create a main.bicep can probably considered independent of this.
